components/sup/src/manager/service/hooks.rs

```diff
@@ -334,36 +344,40 @@ impl Hook for SuitabilityHook {
         let pkg_name = &pkg.name;
         match status.code() {
             Some(0) => {
-                if let Some(reader) = hook_output.stdout() {
-                    if let Some(line_reader) = reader.lines().last() {
-                        match line_reader {
-                            Ok(line) => {
-                                match line.trim().parse::<u64>() {
-                                    Ok(suitability) => {
-                                        outputln!(preamble pkg_name,
-                                                  "Reporting suitability of: {}", suitability);
-                                        return Some(suitability);
-                                    }
-                                    Err(err) => {
-                                        outputln!(preamble pkg_name,
-                                            "Parsing suitability failed: {}", err);
-                                    }
-                                };
-                            }
-                            Err(err) => {
-                                outputln!(preamble pkg_name,
-                                    "Failed to read last line of stdout: {}", err);
-                            }
-                        };
-                    } else {
-                        outputln!(preamble pkg_name,
-                                  "{} did not print anything to stdout", Self::file_name());
+                match hook_output.stdout() {
+                    Ok(reader) => {
+                        if let Some(line_reader) = reader.lines().last() {
+                            match line_reader {
+                                Ok(line) => {
+                                    match line.trim().parse::<u64>() {
+                                        Ok(suitability) => {
+                                            outputln!(preamble pkg_name,
+                                                "Reporting suitability of: {}", suitability);
+                                            return Some(suitability);
+                                        }
+                                        Err(err) => {
+                                            outputln!(preamble pkg_name,
```

Comment:
Why are changes to the suitability hook necessary here?

(but yeah, the method is definitely due for some refactoring love)

Reply:
The suitability hooks are necessary because before we would panic if we could not read the stdout file. This felt a bit extreme to me so we now log an error.

Reply:
That sounds like a good change (not panicking). Let's use this opportunity to get that code under test and then refactor it to something more manageable.